rskj-core/src/main/java/org/ethereum/config/SystemProperties.java

```diff
@@ -705,4 +705,50 @@ private Optional<List<BtcECKey>> getGenesisFederationPublicKeys() {
     public long getGasEstimationCap() {
         return configFromFiles.getLong(PROPERTY_RPC_GAS_ESTIMATION_CAP);
     }
+
+    public DbKind databaseKind() {
+        return DbKind.ofName(configFromFiles.getString("keyvalue.datasource"));
+    }
+
+    public DbKind getDbKindValueFromDbKindLogFile() {
```

Comment:
Im just thinking that we could make this and `generatedDbKindLogFile` methods `static` and probably part of `KeyValueDataSource`, cause `SystemProperties` seems to be more related to node configs rather than to db files?

Reply:
Could you elaborate a bit more? there are some instance methods that are needed here

Reply:
`SystemProperties` provides config values, e.g. the ones defined in `reference.conf`. `getDbKindValueFromDbKindLogFile` method works with database files, that's why I suggested to move it and the other one to the `KeyValueDataSource` class

Reply:
Addressed